### DISCUSSION TITLE
Issues with equivariance testing with larger systems.

### DISCUSSION THREAD

**Discussion by dangthatsright** [e3nn/e3nn#354] [Q&A]
Hi, just wanted to ask about something I've noticed when checking equivariance for some networks. 

The general code I have looks like:

```
from e3nn.nn.models.v2103.gate_points_networks import SimpleNetwork, NetworkForAGraphWithAttributes

net = SimpleNetwork(
    irreps_in="10x0e",
    irreps_out="1x0e",
    max_radius=3.0,
    num_neighbors=5.0,
    num_nodes=5.0
)

def wrapper(pos, x):
    data = dict(pos=pos, x=x)
    return net(data)

num_nodes = 20
pos = torch.randn(num_nodes, 3) 
node_input = net.irreps_in.randn(num_nodes, -1)

assert_equivariant(
    wrapper,
    args_in = [pos, node_input],
    irreps_in=['cartesian_points', net.irreps_in],
    irreps_out=[net.irreps_out],
)
```

This code passes with `num_nodes=20` but the errors start accumulating more and more until it is no longer `equivariant`. This leads me to the question: Do e3nn networks lose their equivariant properties as you increase the number of nodes? Presumably this would happen as you increase the size of the network as well? Are there ways to mitigate this effect?

Thanks!

**Comment by mariogeiger** [comment]
This is due to the increase in the number of neighbors.
To make a measurement more realist (like larger molecules and so on) one can scale the random positions with `sqrt(n)` to keep the minimal distance `O(1)`.

![output](https://user-images.githubusercontent.com/333780/183753524-11150ff3-1659-4bba-b57c-6cfa6f288b75.png)

**Reply by dangthatsright** [reply]
I see. I guess that begs 2 more questions:
1. why do we see this phenomenon?
2. is there a way to get around this? I think it would be interesting to see how networks compare between fewer neighbors + more convolutions and more neighbors + fewer convolutions
